mujoco_warp/_src/collision_driver.py

```diff
@@ -60,8 +61,162 @@ def _zero_collision_arrays(
 
 
 @wp.func
-def _sphere_filter(
+def _plane_filter(
+  size1: float, size2: float, margin1: float, margin2: float, xpos1: wp.vec3, xpos2: wp.vec3, xmat1: wp.mat33, xmat2: wp.mat33
+) -> bool:
+  if size1 == 0.0:
+    # geom1 is a plane
+    dist = wp.dot(xpos2 - xpos1, wp.vec3(xmat1[0, 2], xmat1[1, 2], xmat1[2, 2]))
+    return dist <= size2 + wp.max(margin1, margin2)
+  elif size2 == 0.0:
+    # geom2 is a plane
+    dist = wp.dot(xpos1 - xpos2, wp.vec3(xmat2[0, 2], xmat2[1, 2], xmat2[2, 2]))
+    return dist <= size1 + wp.max(margin1, margin2)
+
+  return True
+
+
+@wp.func
+def _sphere_filter(size1: float, size2: float, margin1: float, margin2: float, xpos1: wp.vec3, xpos2: wp.vec3) -> bool:
+  bound = size1 + size2 + wp.max(margin1, margin2)
+  dif = xpos2 - xpos1
+
+  if size1 != 0.0 and size2 != 0.0:
+    # neither geom is a plane
+    dist_sq = wp.dot(dif, dif)
+    return dist_sq <= bound * bound
+
+  return True
+
+
+@wp.func
+def _aabb_filter(
+  center1: wp.vec3,
+  center2: wp.vec3,
+  size1: wp.vec3,
+  size2: wp.vec3,
+  margin1: float,
+  margin2: float,
+  xpos1: wp.vec3,
+  xpos2: wp.vec3,
+  xmat1: wp.mat33,
+  xmat2: wp.mat33,
+) -> bool:
+  """Axis aligned boxes collision.
+
+  references: mju_raySlab: see Ericson, Real-time Collision Detection section 5.3.3.
+              filterBox: filter contact based on global AABBs.
+  """
+  # get infinite dimensions (planes only)
+  inf1 = wp.vec3i(int(size1[0] >= MJ_MAXVAL), int(size1[1] >= MJ_MAXVAL), int(size1[2] >= MJ_MAXVAL))
+  inf2 = wp.vec3i(int(size2[0] >= MJ_MAXVAL), int(size2[1] >= MJ_MAXVAL), int(size2[2] >= MJ_MAXVAL))
+
+  # if a bounding box is infinite, there must be a collision
+  if (inf1[0] and inf1[1] and inf1[2]) or (inf2[0] and inf2[1] and inf2[2]):
+    return True
+
+  center1 = xmat1 @ center1 + xpos1
+  size1 = xmat1 @ size1
+
+  center2 = xmat2 @ center2 + xpos2
+  size2 = xmat2 @ size2
+
+  margin = wp.max(margin1, margin2)
+  infinite = wp.vec2i(int(inf1[0] or inf1[1] or inf1[2]), int(inf2[0] or inf2[1] or inf2[2]))
```

Comment:
why not move this up so the two infinite checks are together and both early return?

Reply:
updated to remove the infinite checks from the aabb and obb filters. instead, these filters are not called if one of the geoms is a plane (ie, geom_rbound=0)